build-pyg-support.py

```diff
@@ -0,0 +1,283 @@
+#! /usr/bin/env python3
+
+import subprocess
+from threading import Timer
+import os
+import signal
+import tempfile
+import argparse
+import re
+import warnings
+
+PACKAGES = {
+    'torch': {
+        'tag': '1.10.1'
+    },
+    'cluster': {
+        'tag': '*',
+        'url': 'https://github.com/rusty1s/pytorch_cluster.git',
+        'conda_path': '/conda/pytorch-cluster/'
+    },
+    'scatter': {
+        'tag': '2.0.8',
+        'url': 'https://github.com/rusty1s/pytorch_scatter.git',
+        'conda_path': '/conda/pytorch-scatter/'
+    },
+    'sparse': {
+        'tag': '0.6.12',
+        'url': 'https://github.com/rusty1s/pytorch_sparse.git',
+        'conda_path': '/conda/pytorch-sparse/',
+        'metis': False
+    },
+    'spline-conv': {
+        'tag': '*',
+        'url': 'https://github.com/rusty1s/pytorch_spline_conv.git',
+        'conda_path': '/conda/pytorch-spline-conv/'
+    },
+    'pyg': {
+        'tag': '2.0.1',
+        'url': 'https://github.com/pyg-team/pytorch_geometric.git',
+        'conda_path': '/conda/pyg/'
+    }
+}
+
+def run_command(command, timeout=None, run=True):
+    command = 'exec '+command
+    print(command)
+
+    if not run:
+        return
+
+    def kill_proc(process, cancelled):
+        cancelled[0] = True
+        os.killpg(os.getpgid(process.pid), signal.SIGINT)
+
+    process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, start_new_session=True)
+    timer = None
+    cancelled = [False]
+    if timeout is not None:
+        timer = Timer(timeout, kill_proc, [process, cancelled])
+        timer.start()
+    while True:
+        output = process.stdout.readline()
+        if process.poll() is not None:
+            break
+        if output:
+            print(output.strip().decode("utf-8"))
+
+    if timer is not None:
+        timer.cancel()
+    _, stderr = process.communicate()
+
+    if cancelled[0]:
+        warnings.warn("Command timed out: "+command, RuntimeWarning)
+    elif process.returncode != 0:
+        raise RuntimeError(stderr.decode("utf-8"))
+
+def find_file(root, pattern):
+    result = None
+    for name in os.listdir(root):
+        if re.search(pattern, name):
+            if result is not None:
+                raise RuntimeError(
+                    "Regex found multiple matches for pattern '"+pattern+"' in '"+root+"'."
+                )
+            result = name
+
+    if result is None:
+        raise RuntimeError(
+            "Regex found no matches for pattern '"+pattern+"' in '"+root+"'."
+        )
+
+    return result
```

Comment:
`pathlib` has a `glob` function which can be used to find files matching a pattern, so I don't think you need this.

Reply:
Used python's `import glob`

Reply:
Please use https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob instead. `glob` is old.